plugin-scalebar/src/main/java/com/mapbox/maps/plugin/scalebar/ScaleBarImpl.kt

```diff
@@ -252,26 +263,30 @@ class ScaleBarImpl : ScaleBar, View {
     for (i in 0 until pair.second) {
       barPaint.color = if (i % 2 == 0) settings.primaryColor else settings.secondaryColor
       val distanceText = getDistanceText(unitDistance * i)
+      // Make the first text shift to right with borderWidth, the most right text shift to left with INTERNAL_PADDING
+      val xPositionShitForText =
+        if (i == 0) settings.borderWidth else if (i == pair.second - 1) -INTERNAL_PADDING.toFloat() else 0f
+
       textPaint.getTextPath(
-        distanceText, 0, distanceText.length, settings.marginLeft + unitBarWidth * i,
-        settings.textSize + settings.marginTop, path
+        distanceText, 0, distanceText.length, unitBarWidth * i + xPositionShitForText,
```

Comment:
```suggestion
        distanceText, 0, distanceText.length, (unitBarWidth * i) + xPositionShitForText,
```
to make it super obvious